core/src/test/java/org/elasticsearch/index/translog/TranslogTests.java

```diff
@@ -2497,6 +2497,35 @@ public void testMinSeqNoBasedAPI() throws IOException {
         }
     }
 
+    public void testGetSnapshotBetweenAPI() throws IOException {
+        final int operations = randomIntBetween(2, 8096);
+        long seqNo = 0;
+        for (int i = 0; i < operations; i++) {
+            translog.add(new Translog.NoOp(seqNo++, 0, "test'"));
```

Comment:
I think we want this test to be more evil. Right now it’s inserting the sequence numbers into the translog in order. This is divorced from reality; let’s shuffle the sequence numbers. This means you’ll have to change the assertions that you make below.